sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamEnumerableConverter.java

```diff
@@ -118,7 +119,14 @@ public static PipelineOptions createPipelineOptions(Map<String, String> map) {
     for (Map.Entry<String, String> entry : map.entrySet()) {
       args[i++] = "--" + entry.getKey() + "=" + entry.getValue();
     }
-    PipelineOptions options = PipelineOptionsFactory.fromArgs(args).withValidation().create();
+
+    PipelineOptions options = null;
+    try {
+      options = PipelineOptionsFactory.fromArgs(args).withValidation().create();
+    } catch (PipelineOptionUnexpectedPropertyException e) {
+      throw new IllegalArgumentException(e.getUnexpectedPropertyExceptionMessage());
```

Comment:
here you should extract the information from the exception to wrap it or construct a correct message, e.g.:

```java
String message = String.format(" Blah blah ...  RESET", e.getPipelineOptionsClass(), e.getOption(), e.getClosestMatches())
throw new IllegalArgumentException();
```